python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -268,10 +271,17 @@ def _create_repository_using_definitions_args(
     # First, dedupe all definition types.
     sensors = dedupe_object_refs(sensors)
     jobs = dedupe_object_refs(jobs)
-    assets = dedupe_object_refs(assets)
+    assets = _canonicalize_specs_to_assets_defs(dedupe_object_refs(assets))
     schedules = dedupe_object_refs(schedules)
     asset_checks = dedupe_object_refs(asset_checks)
 
+    # add in a default automation condition sensor definition if required
+    sensors = add_default_automation_condition_sensor(
+        sensors,
+        [asset for asset in assets if not isinstance(asset, CacheableAssetsDefinition)],
```

Comment:
does this mean we don't support `CacheableAssetsDefinition` here?

Reply:
Specifically, it means that if you explicitly construct a set of AutomationConditionSensorDefinitions with asset_selections that do not cover a CacheableAssetsDefinition with an automation condition, then that asset would not get automatically targeted by the default_automation_condition_sensor. 

So in order for this to hit you, you'd need to:
- Be using cacheable assets with automation conditions
- Have multiple AutomationConditionSensorDefinitions in your repo
- Neglect to include your cacheable assets definitions in one of those sensors

I consider this enough of an edge case (and one with an easy enough workaround -- just make sure those cacheable assets are in an asset selection), that it's not a big concern

Reply:
Btw @smackesey is about to land https://github.com/dagster-io/dagster/pull/24632. Could this simplify the situation?

Reply:
ooh nice, yes I think I could make use of that

Reply:
Ok will let you sequence this but the less we have to think about `CacheableAssetsDefinition` the better